packages/common/src/types/chainUpdate.ts

```diff
@@ -0,0 +1,226 @@
+    update: TransactionFeeDistribution;
+}
+
+export interface GasRewardsUpdate {
+    updateType: UpdateType.GasRewards;
+    update: GasRewards;
+}
+
+export interface AddAnonymityRevokerUpdate {
+    updateType: UpdateType.AddAnonymityRevoker;
+    update: AddAnonymityRevoker;
+}
+
+export interface AddIdentityProviderUpdate {
+    updateType: UpdateType.AddIdentityProvider;
+    update: AddIdentityProvider;
+}
+
+export interface CooldownParametersUpdate {
+    updateType: UpdateType.CooldownParameters;
+    update: CooldownParameters;
+}
+
+export interface TimeParametersUpdate {
+    updateType: UpdateType.TimeParameters;
+    update: TimeParameters;
+}
+
+export interface ProtocolUpdate {
+    updateType: UpdateType.Protocol;
+    update: ProtocolUpdateDetails;
+}
+
+export interface PoolParametersUpdate {
+    updateType: UpdateType.PoolParameters;
+    update: PoolParameters;
+}
+
+export interface BakerStakeThresholdUpdate {
+    updateType: UpdateType.BakerStakeThreshold;
+    update: BakerStakeThreshold;
+}
+
+export interface Level1Update {
+    updateType: UpdateType.Level1;
+    update: KeyUpdate;
+}
+
+export interface RootUpdate {
+    updateType: UpdateType.Root;
+    update: KeyUpdate;
+}
+
+export type UpdateInstructionPayload =
+    | MicroGtuPerEuroUpdate
+    | EuroPerEnergyUpdate
+    | TransactionFeeDistributionUpdate
+    | FoundationAccountUpdate
+    | MintDistributionUpdate
+    | ProtocolUpdate
+    | GasRewardsUpdate
+    | BakerStakeThresholdUpdate
+    | ElectionDifficultyUpdate
+    | AddAnonymityRevokerUpdate
+    | AddIdentityProviderUpdate
+    | CooldownParametersUpdate
+    | PoolParametersUpdate
+    | TimeParametersUpdate
+    | RootUpdate
+    | Level1Update;
+
+export enum UpdateType {
+    Root = 'root',
+    Level1 = 'level1',
+    Protocol = 'protocol',
+    ElectionDifficulty = 'electionDifficulty',
+    EuroPerEnergy = 'euroPerEnergy',
+    MicroGtuPerEuro = 'microGtuPerEuro',
+    FoundationAccount = 'foundationAccount',
+    MintDistribution = 'mintDistribution',
+    TransactionFeeDistribution = 'transactionFeeDistribution',
+    GasRewards = 'gasRewards',
+    PoolParameters = 'poolParameters',
+    AddAnonymityRevoker = 'addAnonymityRevoker',
+    AddIdentityProvider = 'addIdentityProvider',
+    CooldownParameters = 'cooldownParameters',
+    TimeParameters = 'timeParameters',
+    ProtocolUpdate = 'protocolUpdate',
+    BakerStakeThreshold = 'bakerStakeThreshold',
+    Emergency = 'emergency',
+}
+
+export type KeyUpdate = HigherLevelKeyUpdate | AuthorizationKeysUpdate;
+
+export interface Fraction {
+    numerator: bigint;
+    denominator: bigint;
+}
+
+export type AddIdentityProvider = IpInfo;
+export type AddAnonymityRevoker = ArInfo;
+
+export interface FoundationAccount {
+    address: string;
+}
+
+export interface ProtocolUpdateDetails {
+    message: string;
+    specificationUrl: string;
+    specificationHash: string;
+    specificationAuxiliaryData?: string;
```

Comment:
Why is this optional? Can't an empty string be used?

Reply:
Made it non-optional (empty string is used)